libs/gltfio/src/AssetLoader.cpp

```diff
@@ -343,95 +341,130 @@ FFilamentAsset* FAssetLoader::createInstancedAsset(const uint8_t* bytes, uint32_
     utils::FixedCapacityVector<uint8_t> glbdata(byteCount);
     std::copy_n(bytes, byteCount, glbdata.data());
 
+    // The ownership of an allocated `sourceAsset` will be moved to FFilamentAsset::mSourceAsset.
     cgltf_data* sourceAsset;
     cgltf_result result = cgltf_parse(&options, glbdata.data(), byteCount, &sourceAsset);
     if (result != cgltf_result_success) {
         slog.e << "Unable to parse glTF file." << io::endl;
         return nullptr;
     }
 
-    createRootAsset(sourceAsset);
+    FFilamentAsset* fAsset = createRootAsset(sourceAsset);
     if (mError) {
-        delete mAsset;
-        mAsset = nullptr;
+        delete fAsset;
+        fAsset = nullptr;
```

Comment:
nit: can be omitted IMO.  Same for similar lines below.

Reply:
Thanks for the suggestion! I agree, we can improve the way of handling errors during asset loading in a separate commit.